tools/twix/src/twix_painter.rs

```diff
@@ -4,55 +4,48 @@ use std::{
 };
 
 use eframe::{
-    egui::{Painter, Response, Sense, Ui},
+    egui::{pos2, Context, Painter, Response, Sense, TextureId, Ui},
     emath::{Pos2, Rect},
     epaint::{Color32, PathShape, Shape, Stroke},
 };
-use nalgebra::{Rotation2, SMatrix, Similarity2};
+use nalgebra::{Affine2, Rotation2, SMatrix, Similarity2};
 
-use coordinate_systems::{Field, Ground};
-use geometry::{arc::Arc, circle::Circle, direction::Direction};
-use linear_algebra::{point, vector, IntoTransform, Isometry2, Point2, Pose2, Vector2};
+use coordinate_systems::{Field, Ground, Screen};
+use geometry::{arc::Arc, circle::Circle, direction::Direction, rectangle::Rectangle};
+use linear_algebra::{point, vector, IntoTransform, Isometry2, Point2, Pose2, Transform, Vector2};
 use types::{field_dimensions::FieldDimensions, planned_path::PathSegment};
 
-#[derive(Clone)]
-pub enum CoordinateSystem {
-    RightHand,
-    LeftHand,
-}
-
-impl CoordinateSystem {
-    fn y_scale(&self) -> f32 {
-        match self {
-            CoordinateSystem::RightHand => -1.0,
-            CoordinateSystem::LeftHand => 1.0,
-        }
-    }
-}
-
+type ScreenTransform<Frame> = Transform<Frame, Screen, Affine2<f32>>;
 pub struct TwixPainter<Frame> {
     painter: Painter,
     pixel_rect: Rect,
-    world_to_pixel: Similarity2<f32>,
-    camera_coordinate_system: CoordinateSystem,
+    world_to_pixel: ScreenTransform<Frame>,
     frame: PhantomData<Frame>,
 }
+trait Convert {
```

Comment:
please insert newlines between struct and trait definitions

Reply:
done